conans/test/functional/profiles/profiles_test.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+
+
+class ProfileTest(unittest.TestCase):
+
+    def test_env_var_with_variable_substitution(self):
+        client = TestClient()
+        profile = """
```

Comment:
Please ``textwrap.dedent()`` and indent the contents with respect to the ``profile`` variable definition.